contracts/MerkleDelegation.sol

```diff
@@ -10,57 +10,42 @@ struct DelegatorRecord {
     uint256 blockNumber;
 }
 
-struct DelegateeRecord {
-    address delegatee;
-    uint256 percentage; // 100,000 is 100%, 1,000 is 1%.
-}
-
 // https://docs.openzeppelin.com/contracts/4.x/api/access#Ownable
 // https://docs.openzeppelin.com/contracts/4.x/api/security#Pausable
 // https://docs.openzeppelin.com/contracts/4.x/api/utils#MerkleProof
 contract MerkleDelegation is Ownable, Pausable {
-    address public governanceToken;
+    address public immutable governanceToken;
     mapping(address => DelegatorRecord) public delegation;
     mapping(address => address) public delegateToDelegator;
 
-    event OwnerInitialized(address owner);
-    event OwnerChanged(address newOwner);
     event SetDelegates(
         address indexed delegator,
         bytes32 indexed trieRoot,
         uint256 blockNumber
     );
-    event ClearDelegate(address indexed delegator, bytes32 trieRoot);
 
     constructor(address _governanceToken) Ownable() Pausable() {
-        require(
-            _governanceToken != address(0),
-            "DR: token addr must be non-zero"
-        );
+        require(_governanceToken != address(0), "token addr must be non-zero");
         governanceToken = _governanceToken;
     }
 
-    //TODO: also the owner/admin can update the governance token address that is hashed with all leafs.
-    // function updateGovernanceToken(address newGovernanceToken) { governanceToken = newGovernancetoken; }
-
     function setDelegateTrie(address delegator, bytes32 trieRoot)
         external
         whenNotPaused
     {
-        require(delegator != address(0), "DR: delegator must be non-zero");
-        require(trieRoot != bytes32(0), "DR: trieRoot must be non-zero");
-        require(msg.sender == delegator, "DR: delegator must be msg.sender");
-        // Record to storage.
-        delegation[delegator].trieRoot = trieRoot;
-        delegation[delegator].blockNumber = block.number;
-        emit SetDelegates(delegator, trieRoot, block.number);
+        require(delegator == msg.sender, "delegator must be msg.sender");
+        require(trieRoot != bytes32(0), "trieRoot must be non-zero");
+        _setDelegateTrie(delegator, trieRoot);
     }
 
     function clearDelegateTrie(address delegator) external whenNotPaused {
-        require(delegator != address(0), "DR: delegator must be non-zero");
-        require(msg.sender == delegator, "DR: delegator must be msg.sender");
-        // Remove delegator record to save storage gas.
-        delete delegation[delegator];
+        require(delegator == msg.sender, "delegator must be msg.sender");
+        _setDelegateTrie(
+            delegator,
+            bytes32(
+                0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421
```

Comment:
I would leave a comment to indicate that `0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421` is the root of an empty merkletrie rather than some magic hash value.

Reply:
Nice, thank you! Addressed in [ad46fca](https://github.com/windranger-io/windranger-mass-delegation-contracts/pull/17/commits/ad46fcad2d4126ea31e7c07a32937a5f2a84142f)